# 51) Etki Büyüklüğü (Effect Size - Cohen's d)
Konu 50'de (P Değeri) önemli bir uyarı yapmıştık: **p-değeri, etkinin ne kadar "büyük/önemli" olduğunu göstermez** sadece ne kadar "şaşırtıcı/nadir" olduğunu gösterir. Effect Size, tam olarak bu boşluğu dolduruyor: **etkinin gerçek büyüklüğünü, standartlaştırılmış bir şekilde ölçüyor.**

## Neden P-Değeri Yetmez? (Kritik Motivasyon)
Çok büyük bir örneklemle (örn: n=1 milyon), gerçekte **önemsiz derecede küçük** bir fark bile, çok küçük bir p-değeri (p<0.001 gibi) verebilir çünkü büyük örneklem, standart hatayı küçültür, bu da küçük farkları bile "istatistiksel olarak anlamlı" gösterebilir. Ama bu fark, **iş açısından gerçekten önemli mi**? P-değeri tek başına bunu söylemez — Effect Size söyler.

## Cohen's d Formülü (İki Grup Karşılaştırması İçin)
$$d = \frac{\bar{x}_1 - \bar{x}_2}{s_{pooled}}$$

Burada $\bar{x}_1 - \bar{x}_2$, iki grubun ortalama farkı (Konu 29'da gördüğümüz), $s_{pooled}$ ise iki grubun **birleşik (pooled) standart sapması** (Konu 29'da "İki Popülasyon Ortalama Farkı" konusunda hesapladığımız $s_p$'nin karekökü).

**Sezgisel anlamı:** "İki grup arasındaki fark, kaç standart sapma büyüklüğünde?" tıpkı Z-Score'un (Konu 20) "bir değer, ortalamadan kaç std uzakta" sorusuna benzer bir mantık, ama burada iki grup ortalaması arasındaki farka uygulanıyor.

## Yorumlama (Cohen'in Önerdiği Kaba Eşikler)
- **d ≈ 0.2** → Küçük etki
- **d ≈ 0.5** → Orta etki
- **d ≈ 0.8** → Büyük etki

Bu eşikler kesin kurallar değil, genel kabul görmüş **kaba referans noktalarıdır** bağlama göre yorumlanmalıdır.

## P-Değeri ile Birlikte Kullanım
İkisi birbirini tamamlar, biri diğerinin yerini tutmaz:
- **P-değeri** → "Bu fark gerçek mi, yoksa şans eseri mi?" (anlamlılık)
- **Effect Size** → "Bu fark ne kadar büyük/önemli?" (büyüklük)

İyi bir analiz, ikisini birlikte raporlar: "Fark istatistiksel olarak anlamlıydı (p<0.05) VE pratik olarak da önemli bir büyüklükteydi (d=0.6)" gibi. Sadece p-değerine bakıp karar vermek, büyük örneklemli çalışmalarda yanıltıcı sonuçlara yol açabilir.

In [1]:
import numpy as np
from scipy import stats

# --- Senaryo: Genç ve Yaşlı Kullanıcıların Uygulamada Geçirdiği Süre (dakika) ---
genc_grup = np.array([45, 52, 38, 60, 55, 48, 42, 50, 58, 47])
yasli_grup = np.array([30, 35, 28, 40, 33, 38, 32, 36, 29, 34])

# --- Hipotezler ---
print("H0: Genç ve yaşlı kullanıcıların uygulamada geçirdiği ortalama süre AYNIDIR.")
print("H1: Genç ve yaşlı kullanıcıların uygulamada geçirdiği ortalama süre FARKLIDIR.")
print("-" * 60)

# --- Betimsel Bakış ---
print(f"Genç grup ortalaması: {np.mean(genc_grup)}")
print(f"Yaşlı grup ortalaması: {np.mean(yasli_grup)}")

# --- P-Değeri (Bağımsız İki Örneklem T Testi - ileride Konu 57'de detaylı göreceğiz) ---
t_stat, p_degeri = stats.ttest_ind(genc_grup, yasli_grup)
print(f"\nt-istatistiği: {t_stat:.4f}")
print(f"p-değeri: {p_degeri:.10f}")

alpha = 0.05
if p_degeri < alpha:
    print(f"p-değeri ({p_degeri:.6f}) < α ({alpha}) → H0 reddedilir, fark istatistiksel olarak anlamlıdır.")
else:
    print(f"p-değeri ({p_degeri:.6f}) >= α ({alpha}) → H0 reddedilemez.")

# --- Cohen's d (Effect Size) ---
n1, n2 = len(genc_grup), len(yasli_grup)
s1, s2 = np.std(genc_grup, ddof=1), np.std(yasli_grup, ddof=1)
s_pooled = np.sqrt(((n1-1)*s1**2 + (n2-1)*s2**2) / (n1+n2-2))
cohens_d = (np.mean(genc_grup) - np.mean(yasli_grup)) / s_pooled

print(f"\nCohen's d: {cohens_d:.4f}")
if abs(cohens_d) < 0.2:
    yorum = "Küçük"
elif abs(cohens_d) < 0.5:
    yorum = "Orta"
else:
    yorum = "Büyük"
print(f"Etki büyüklüğü yorumu: {yorum} etki")

H0: Genç ve yaşlı kullanıcıların uygulamada geçirdiği ortalama süre AYNIDIR.
H1: Genç ve yaşlı kullanıcıların uygulamada geçirdiği ortalama süre FARKLIDIR.
------------------------------------------------------------
Genç grup ortalaması: 49.5
Yaşlı grup ortalaması: 33.5

t-istatistiği: 6.3411
p-değeri: 0.0000056474
p-değeri (0.000006) < α (0.05) → H0 reddedilir, fark istatistiksel olarak anlamlıdır.

Cohen's d: 2.8358
Etki büyüklüğü yorumu: Büyük etki


### Sonuç
Genç ve Yaşlı insanların uygulamada geçirdiği sürelerin ortalamaları arasındaki farkın istatistiksel olarak anlamlı olup olmadığını keşfetmek adına bağımsız örneklem t-testi uyguladık (Konu 57'de değineceğiz). Bu testin sonucuna göre Genç grubun uygulamada geçirdiği ortalama süre yaşlı insanların oluşturduğu grubun uygulamada geçirdiği ortalama süreye göre oldukça yüksek ve istatistiksel olarak anlamlıdır (Mgenç - Myaşlı = 16, p < .001) ayrıca bu farka yönelik etki büyüklüğü de büyük düzeydedir (cohen's d=2.84).